text/0000-road-to-typescript.md

```diff
@@ -0,0 +1,160 @@
+
+- **Supported compilation modes.** What should the out-of-the-box settings be for compiling Ember apps and addons written in TypeScript? This will need to be written with an eye to Embroider and the v2 Package Format as end goals, but without assuming them as available at the starting point. It must also address the transition path from current ember-cli-typescript (if any). It *must* address interoperation hazards between Babel and TypeScript: for example, in the decorators implementations, use of the `isolateModules` setting and associated lints, etc.
+
+- **The `@glimmer/component@2.0.0` Type API.** The current Glimmer component TypeScript API works well, but research efforts into “typed templates” and [component documentation](https://github.com/emberjs/rfcs/pull/678) have both exposed the importance of being able to specify more “type parameters” for components than simply its args, including its root element(s) and blocks, if any. Additionally, the types should be readily extensible if other such concerns emerge in the future.
+
+- **Official Documentation.** It is important, if Ember officially supports TypeScript, that we support it in all of Ember’s documentation, including guides, API docs, etc. This requires considerable design effort, including thinking through how to present them (side by side? toggle-able? etc.), where to introduce discussion of TypeScript support in the guides, etc.
+
+- **How do consumers generate TypeScript apps and addons (and blueprints more generally)?** Today, users install `ember-cli-typescript` to opt into using TypeScript, and there is a wholly separate (and badly-maintained!) set of blueprints for working with TypeScript. Should we support `ember new --typescript`? How should blueprints work?
+
+There are also concerns which must be addressed, but do not require RFCs (they are key implementation details, but *are* implementation details):
+
+- **How are the published types generated?** While much of Ember is implemented in TypeScript, its internal types are not currently ready to be used for public API, and it will likely take some time for them to be ready (with different packages ready at different times).
+
+- **How does the build tooling publish the types?** TypeScript generally assumes that types are published with packages of the same name. While there is [now](https://emberjs.github.io/rfcs/0706-deprecate-ember-global.html) a path to publishing actual packages like `"@ember/object"`, this effort should not be gated on any such effort, and there is additional work to be done. However, that work simply consists of implementation details, rather than defining new public API.
```

Comment:
Same question here: Is this about `@ember/*` packages only? Or generic section about all packages, such as community packages?

Reply:
Same answer as above. 😉